lib/ash/filter/filter.ex

```diff
@@ -235,7 +235,10 @@ defmodule Ash.Filter do
         filter
 
       {:error, error} ->
-        raise Ash.Error.to_error_class(error)
+        raise Ash.Error.to_error_class(error,
+                error_context:
+                  "parsing #{inspect(resource)} #{inspect(statement)} #{inspect(aggregates)} #{inspect(calculations)} in #{inspect(context)}"
```

Comment:
I am not sure what actual context we want in some of these call sites.  This is just an example, I doubt this is useful as is.  Will need some guidance on these.

Reply:
I think just the resource and statement is great! :)